### PR TITLE
Documentation: add expected nmodl units 

### LINKED ISSUE #1753: Document expected units used for each nmodl quantity
The units expected for the conductance, voltage, generated currents, ion concentrations, time, etc are not obvious in nmodl. 
A table of these units should be added to the docs.

### PR DESCRIPTION
**Pull request by noraabiakar**
Fixes #1753

### CODE REVIEW

File: doc/fileformat/nmodl.rst [hunk 1]
```diff
@@ -29,6 +29,35 @@ does not allow some constructions used in NEURON's NMODL.
...
+===============================================  ===================================================  ==========
+quantity                                         identifier                                           unit
+===============================================  ===================================================  ==========
+voltage                                          v / v_peer                                           mV
+time                                             t                                                    ms
+temperature                                      celsius                                              °C
+diameter (cross-sectional)                       diam                                                 µm
+
+current_density (density mechanisms)             identifier defined using ``NONSPECIFIC_CURRENT``     mA/cm²
+conductivity (density mechanisms)                identifier inferred from current_denisty equation    S/cm²
```
Comment (schmitts) [inline doc/fileformat/nmodl.rst:47]: typo: denisty

**Review by schmitts — CHANGES_REQUESTED** [review]
Apart from the minor fixups, Arbor experts need to check the units

**Review by bcumming — APPROVED** [review]
lgtm

**Comment by jlubo** [comment]
Thanks! This will help a lot reproducing results from other simulations.

**Pull request merged**